File: src/main/java/com/example/demo/constants/ApplicationConstants.java
package com.example.demo.constants;

import lombok.experimental.UtilityClass;

/**
 * Common path constants.
 */
@UtilityClass
public class ApplicationConstants {
    public final String ROOT = "/";
    public final String REDIRECT = "redirect:";
    public final String USERNAME_CHECK = "/userNameCheck";
    public final String USER_CHECK = "/userCheck";
    public final String CONTENT = "/content";
    public final String ALL = "/**";
    public final String MAIN = "/main";
    public final String LOGIN = "/login";
    public final String REGISTER = "/register";

    @UtilityClass
    public class RegisterConstants {
        public final String CHARACTERS = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789";
        public final int PASSWORD_LENGTH = 15;
    }
}

File: src/main/kotlin/com/example/demo/constants/ApplicationConstants.kt
package com.example.demo.constants

/**
 * アプリケーション内で使用する共通のパス定数
 */
object ApplicationConstants {
    // ルート（トップページ）パス
    const val ROOT = "/"
    const val HOME = "/home"
    const val REDIRECT = "redirect:"
    const val USERNAME_CHECK = "/userNameCheck"
    const val USER_CHECK = "/userCheck"
    const val CONTENT = "/content"
    const val ALL = "/**"
    const val MAIN = "/main"
    const val LOGIN = "/login"
    const val REGISTER = "/register"

    /**
     * 会員登録に関する定数
     */
    object RegisterConstants {
        // パスワードに使用する文字集合
        const val CHARACTERS = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789"

        // 生成パスワード長の既定値
        const val PASSWORD_LENGTH = 15
    }
}

